src/System.Private.CoreLib/shared/System/Number.Parsing.cs

```diff
@@ -1728,6 +1755,10 @@ internal static unsafe bool TryParseSingle(ReadOnlySpan<char> value, NumberStyle
             {
                 ReadOnlySpan<char> valueTrim = value.Trim();
 
+                // This code would be simpler if we only had the concept of `InfinitySymbol`, but
```

Comment:
The logic would have been so much simpler if we just had a singular `InfinitySymbol` and negative infinity was computed as `NegativeSign + InfinitySymbol`